### PR TITLE
DCS-1248 Persisting additional conditions version

### PR DESCRIPTION
**Pull request by andrewrlee**
Renaming conditions version to additional conditions version

Migrating existing licences with additional conditions to the existing version

### CODE REVIEW

File: migrations/20211123172816_set-additional-conditions-version.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  await knex.schema.raw(`
+    update licences set additional_conditions_version = 1
+    where licence  -> 'licenceConditions' -> 'additional' is not null
+`)
+}
+
+exports.down = async function down(knex) {
+  await knex.schema.raw(`
+  update licences set additional_conditions_version = null
+  where licence  -> 'licenceConditions' -> 'additional' is not null
```
Comment (bell-pa) [inline migrations/20211123172816_set-additional-conditions-version.js:11]: Trivial observation: Don't think you need a restriction all additional_conditions_version were null before 'up'

**Review by GurnankCheema — DISMISSED** [review]
LGTM

File: test/routes/licenceConditions.test.ts [hunk 2]
```diff
@@ -137,7 +137,7 @@ describe('/hdc/licenceConditions', () => {
     })
   })
 
-  describe('POST /hdc/licenceConditions/:section/:bookingId', () => {
+  describe('POST /hdc/licenceConditions/standard/:bookingId', () => {
```
Comment (bell-pa) [inline test/routes/licenceConditions.test.ts:140]: Doesn't look right at line 185 below. This code:

``
licenceService.getLicence.mockResolvedValue({
          licence: { key: 'value' },
          versionDetails: { conditions_version: 1 },
        })
``

Still has `conditions_version` should that be `additional_conditions_version`?

File: test/routes/licenceConditions.test.ts [hunk 3]
```diff
@@ -250,25 +250,73 @@ describe('/hdc/licenceConditions', () => {
...
+    ]
+
+    routes.forEach((route) => {
+      test(`redirects to '${route.nextPath}' page for ${route.url}`, () => {
         const licenceService = createLicenceServiceStub()
         licenceService.getLicence.mockResolvedValue({
           licence: { key: 'value' },
-          versionDetails: { conditions_version: undefined },
+          versionDetails: { conditions_version: 1 },
         })
```
Comment (bell-pa) [inline test/routes/licenceConditions.test.ts:298]: `additional_conditions_version`?

File: test/routes/licenceConditions.test.ts [hunk 4]
```diff
@@ -250,25 +250,73 @@ describe('/hdc/licenceConditions', () => {
...
+          .post(route.url)
+          .send(route.body)
           .expect(302)
-          .expect(() => {
-            expect(licenceService.setConditionsVersion).toHaveBeenCalledWith(2, 1)
+          .expect((res) => {
+            expect(licenceService.updateLicenceConditions).toHaveBeenCalledWith(
+              1,
+              { licence: { key: 'value' }, versionDetails: { conditions_version: 1 } },
+              {},
```
Comment (bell-pa) [inline test/routes/licenceConditions.test.ts:310]: Should this be `additional_conditions_version`? See licenceService.ts LicenceRecord interface

**Review by bell-pa — CHANGES_REQUESTED** [review]
Think you've missed converting conditions_version to additional_conditions_version in some places.

File: migrations/20211123172816_set-additional-conditions-version.js (hunk 1 above)
Comment (andrewrlee) [reply migrations/20211123172816_set-additional-conditions-version.js]: I've removed the `where`

File: test/routes/licenceConditions.test.ts (hunk 2 above)
Comment (andrewrlee) [reply test/routes/licenceConditions.test.ts:140]: fixed these

File: test/routes/licenceConditions.test.ts (hunk 4 above)
Comment (andrewrlee) [reply test/routes/licenceConditions.test.ts:310]: fixed these

File: test/routes/licenceConditions.test.ts (hunk 3 above)
Comment (andrewrlee) [reply test/routes/licenceConditions.test.ts:298]: fixed these

**Review by bell-pa — APPROVED** [review]
LGTM

**Pull request merged**